packages/shared/src/components/post/PostActions.tsx

```diff
@@ -77,6 +81,31 @@ export function PostActions({
     await toggleDownvote({ payload: post, origin });
   };
 
+  const [getPost, updatePost] = useUpdateQuery(
+    postByIdQueryOptions({
+      id: post.id,
+    }),
+  );
+
+  useQuerySubscription(({ variables }) => {
+    const { type, entityId } = variables;
+
+    if (type === 'POST') {
+      if (entityId !== post.id) {
+        return;
+      }
+
```

Comment:
I'm still wondering if some of this logic could be automated?
EG only passing the actual change and underlying logic can determine what the initial data is based on query key?

Reply:
Technically it can, if we split that each award mutation has different key then we can just update, currently all awards are under same mutation/key and this is why it has some additional checks.